src/interpreter.ts

```diff
@@ -699,7 +699,7 @@ function evalNode(node: SyntaxNodeRef, input: string, args: any[]) {
     }
 
     // a[1]
-    if (filterFn.type === 'number') {
+    if (filterFn.type === 'number' || typeof(filterFn(context)) == 'number') {
```

Comment:
Maybe what needs to get fixed is `filterFn.type` as it was coming back as `undefined`? But maybe that's not possible if the filter function needs to be executed with the context to determine its type.